### PR TITLE
bug/16347 Display DoB for duplicate names in generated pins page

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
* If pupil has same first name and surname, then date of birth is also displayed 
* If pupil has same first name, surname name and same date of birth, then middle name is also displayed

### CODE REVIEW

File: admin/spec/service/pupil.service.spec.js [hunk 1]
```diff
@@ -65,4 +66,19 @@ describe('pupil service', () => {
...
     })
   })
+  describe('displayDoB', () => {
+    it('has showDoB property as true for pupils with same fullname', () => {
+      const pupil1 = Object.assign({}, pupilMock)
+      const pupil2 = Object.assign({}, pupilMock)
+      const pupils = pupilService.displayDoB([pupil1, pupil2])
+      expect(pupils[0].showDoB && pupils[1].showDoB).toBeTruthy()
+    })
+    it('has showDoB property as fa;se for pupils with different fullname', () => {
```
Comment (GuyHarwood) [inline admin/spec/service/pupil.service.spec.js]: missing l in false :)

Comment (GeorgeChatzigiannis) [reply admin/spec/service/pupil.service.spec.js]: @GuyHarwood updated

File: admin/services/pupil.service.js [hunk 2]
```diff
@@ -87,6 +87,28 @@ const pupilService = {
       }
     }
     return true
+  },
+  /**
+   * Adds show Date of Birth flag for pupils with equal full names
```
Comment (jon-shipley) [inline admin/services/pupil.service.js]: There should be a note to say this *assumes* the array of objects have been sorted by some criteria.

File: admin/services/pupil.service.js [hunk 3]
```diff
@@ -87,6 +87,28 @@ const pupilService = {
...
+   * @param {Array} pupils
+   * @returns {Array}
+   */
+  requiresExtraFields: (pupils) => {
```
Comment (jon-shipley) [inline admin/services/pupil.service.js]: This name seems a bit vague to me.  Perhaps something like `addIdentificationFlags` is more descriptive?

**Pull request merged**